src/Common/Admin/Entities/Link.php

```diff
@@ -33,19 +33,33 @@ class Link extends Base_Entity {
 	 */
 	private string $text = '';
 
+	/**
+	 * Content for the link if not a string.
+	 *
+	 * @var ?Base_Entity
+	 */
+	private ?Base_Entity $content = null;
+
 	/**
 	 * Link constructor.
 	 *
 	 * @since 6.1.0
 	 *
-	 * @param string      $url        The URL for the link.
-	 * @param string      $text       The text for the link.
-	 * @param ?Classes    $classes    The classes for the link.
-	 * @param ?Attributes $attributes The attributes for the link.
+	 * @param string             $url        The URL for the link.
+	 * @param string|Base_Entity $content    The text or entity for the link.
+	 * @param ?Classes           $classes    The classes for the link.
+	 * @param ?Attributes        $attributes The attributes for the link.
 	 */
-	public function __construct( string $url, string $text, ?Classes $classes = null, ?Attributes $attributes = null ) {
-		$this->url  = $url;
-		$this->text = $text;
+	public function __construct( string $url, $content, ?Classes $classes = null, ?Attributes $attributes = null ) {
+		$this->url = $url;
+
+		if ( is_string( $content ) ) {
+			$this->text = $content;
+		}
+
+		if ( $content instanceof Base_Entity ) {
+			$this->content = $content;
+		}
```

Comment:
This set of checks for `$content` aren't strict enough. If something other than a `string` or `Base_Entity` is passed, then we don't have a check in place that covers that situation.

What I propose is that we allow for only `string` or `Base_Entity`, and convert a `string` into a `Plain_Text` object to handle compatibility with what is already written.